mlperf_logging/rcp_checker/rcp_checker.py

```diff
@@ -37,64 +38,83 @@
     '''
     Extract convergence epochs (or train_samples for BERT)
     from a list of submission files
-    Returns the batch size and the list of epochs (or samples) to converge
-    -1 means run did not converge. Return None if > 1 files
-    fail to converge
+    Returns the batch size, the list of epochs (or samples) to converge, and benchmark name
+    bs of -1 means batch size could not be determined. subm_epochs is None
+    if > 1 files fail to converge.
     '''
     not_converged = 0
     subm_epochs = []
     bs = -1
-    use_train_samples = benchmark == 'bert' and bert_train_samples
+    benchmark = None
     for result_file in result_files:
-        with open(result_file, 'r', encoding='latin-1') as f:
-            file_contents = f.readlines()
-            use_eval_error = _detect_eval_error(file_contents)
-            for line in file_contents:
-                if TOKEN not in line:
-                    continue
-                line = re.sub(".*"+TOKEN, TOKEN, line).strip()
-                if line.startswith(TOKEN):
-                    str = line[len(TOKEN):]
-                    if "global_batch_size" in str:
-                        # Do we need to make sure global_batch_size is the same
-                        # in all files? If so, this is obviously a bad submission
-                        bs = json.loads(str)["value"]
-                    if not use_train_samples and (((not use_eval_error) and "eval_accuracy" in str) or
-                                                  (use_eval_error and "eval_error" in str)):
-                        eval_accuracy_str = str
-                        conv_epoch = json.loads(eval_accuracy_str)["metadata"]["epoch_num"]
-                        conv_epoch = round(conv_epoch, 3)
-                    if use_train_samples and "train_samples" in str:
-                        eval_accuracy_str = str
-                        conv_epoch = json.loads(eval_accuracy_str)["value"]
-                    if "run_stop" in str:
-                        # Epochs to converge is the the last epochs value on
-                        # eval_accuracy line before run_stop
-                        conv_result = json.loads(str)["metadata"]["status"]
-                        if conv_result == "success":
-                            subm_epochs.append(conv_epoch)
-                        else:
-                            subm_epochs.append(1e9)
-                            not_converged = not_converged + 1
+        curr_not_converged, curr_subm_epochs, curr_bs, curr_benchmark = read_submission_file(result_file, bert_train_samples)
+
+        subm_epochs.append(curr_subm_epochs)
+        not_converged += curr_not_converged
+
+        if bs == -1:
+            bs = curr_bs
+        else:
+            if curr_bs != bs:
+                logging.warning(' Batch sizes in files do not match.')
+                return -1, None, None
+
+        if benchmark is None:
```

Comment:
it seems benchmark is always going to be None here since it is initialized to None in line 92, is this intended? is the benchmark variable still useful here?